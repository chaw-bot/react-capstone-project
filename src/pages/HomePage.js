import React from 'react';
import { Link } from 'react-router-dom';
import { useSelector } from 'react-redux';

const HomePage = () => {
  const ContinentStore = useSelector((store) => store.continent);

  return (
    <div className="ContinentPage" id={ContinentStore}>
      <h1 className="continent">World Wide</h1>
      <ul className="dataUL">
        {
        ContinentStore.map((continent) => (
          <Link key={continent.name} to={{ pathname: `/continent/${continent.name}` }}>
            <li className="ContinentDetails">
              <div className="titles">
                <h1 className="continentName">
                  {continent.name}
                </h1>
                <h2 className="countryNumbers">
                  {continent.number}
                  {' '}
                  countries
                </h2>
              </div>
              <div className="map-div">
                <img src={continent.map} alt="Continent map" className={`map ${continent.classname}`} />
              </div>
            </li>
          </Link>
        ))
      }
      </ul>
    </div>
  );
};

export default HomePage;
