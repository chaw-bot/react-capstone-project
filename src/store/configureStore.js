import { combineReducers, applyMiddleware, createStore } from 'redux';
import thunk from 'redux-thunk';
import logger from 'redux-logger';
import { StatsReducer } from './reducer';
import ContinentReducer from './continentReducer';

const reducers = combineReducers({
  details: StatsReducer,
  continent: ContinentReducer,
});

const middlewares = [thunk, logger];

const store = createStore(
  reducers,
  applyMiddleware(...middlewares),
);

export default store;
