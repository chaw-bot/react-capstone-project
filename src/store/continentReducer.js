import Africa from '../images/Africa.png';
import asia from '../images/asia.png';
import australia from '../images/australia.png';
import europe from '../images/europe.png';
import northamerica from '../images/northamerica.png';
import southamerica from '../images/southamerica.png';

const intialState = [
  {
    name: 'Africa',
    classname: 'Africa',
    map: Africa,
    number: 58,
  },
  {
    name: 'Asia',
    classname: 'Asia',
    map: asia,
    number: 49,
  },
  {
    name: 'Australia-Oceania',
    classname: 'Australia-Oceania',
    map: australia,
    number: 14,
  },
  {
    name: 'Europe',
    classname: 'Europe',
    map: europe,
    number: 48,
  },
  {
    name: 'North America',
    classname: 'North-America',
    map: northamerica,
    number: 39,
  },
  {
    name: 'South America',
    classname: 'South-America',
    map: southamerica,
    number: 14,
  },
];

const ContinentReducer = (state = intialState) => state;

export default ContinentReducer;
