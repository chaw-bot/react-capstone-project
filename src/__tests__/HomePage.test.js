import {
  render, waitFor, screen, act,
} from '@testing-library/react';
import { BrowserRouter as Router, Route, Routes } from 'react-router-dom';
import '@testing-library/jest-dom';
import { Provider } from 'react-redux';
import store from '../store/configureStore';
import HomePage from '../pages/HomePage';
import FetchStats from '../store/api';

jest.mock('../store/api');

describe('Render HomePAge', () => {
  beforeEach(() => {
    FetchStats.mockResolvedValue(
      [
        {
          continent: 'Africa',
          country: 'Botswana',
          country_flag: 'https://disease.sh/assets/img/flags/af.png',
          country_id: 4,
          population: 40135620,
          todays_cases: 17,
          todays_deaths: 0,
          todays_recovered: 55,
          total_active: 13320,
          total_cases: 156812,
          total_deaths: 7361,
          total_recovered: 136131,
          total_tests: 782926,
        },
      ],
    );
  });

  test('Test shows countries', async () => {
    await act(async () => {
      render(
        <Provider store={store}>
          <Router>
            <Routes>
              <Route exact path="/" element={<HomePage />} />
            </Routes>
          </Router>
        </Provider>,
      );
    });

    await waitFor(() => expect(screen.getByText('Botswana')).toBeInTheDocument());
  });
});
