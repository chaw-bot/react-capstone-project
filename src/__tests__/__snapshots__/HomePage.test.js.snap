// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render components render NavBar 1`] = `
<nav>
  <div
    className="nav1"
  >
    <a
      aria-current="page"
      className="active"
      exact="true"
      href="/"
      onClick={[Function]}
    >
      
    </a>
    <h1
      className="header1"
    >
      CoviData
    </h1>
  </div>
  <div
    className="nav2"
  >
    <img
      alt="logo"
      className="logo"
      src="logo.png"
    />
    <h1
      className="header2"
    >
      Corona Afrometer
    </h1>
  </div>
</nav>
`;

exports[`Render components render NavBar 2`] = `
<ul
  className="dataUL"
/>
`;
